### PR TITLE
Update newbs_building_firmware.md

### PR DESCRIPTION
**Pull request by jacobjeffery**
Fixed all of the .c references to .json

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/newbs_building_firmware.md [hunk 1]
```diff
@@ -30,11 +30,11 @@ Look at the output from that command, you should see something like this:
...
     Ψ Created a new keymap called <github_username> in: /home/me/qmk_firmware/keyboards/clueboard/66/rev3/keymaps/<github_username>.
 
-This is the location of your new `keymap.c` file.
+This is the location of your new `keymap.json` file.
 
-## Open `keymap.c` In Your Favorite Text Editor
+## Open `keymap.json` In Your Favorite Text Editor
 
-Open your `keymap.c` file in your text editor. Inside this file you'll find the structure that controls how your keyboard behaves. At the top of `keymap.c` there may be some defines and enums that make the keymap easier to read. Farther down you'll find a line that looks like this:
+Open your `keymap.json` file in your text editor. Inside this file you'll find the structure that controls how your keyboard behaves. At the top of `keymap.c` there may be some defines and enums that make the keymap easier to read. Farther down you'll find a line that looks like this:
```
Comment (jacobjeffery) [inline docs/newbs_building_firmware.md:37]: I missed the second .c reference in this paragraph.

**Review by zvecr — CHANGES_REQUESTED** [review]
The majority of keyboards still have the default keymap implemented with `keymap.c`, which is then used by `qmk new-keymap`. Also, the text blocks after these changes, still refer to typical `keymap.c` logic.

The change as it is right now, is more likely to confuse new users.

**Comment by jacobjeffery** [comment]
Gotcha,. Thank you @zvecr. Admittedly I'm may have moved a bit too quickly here.

**Comment by zvecr** [comment]
Closing for now as we will rework this area post #23451.

**Pull request closed without merging**